From daad6459721cb1b3155bd819ba7eac2963256153 Mon Sep 17 00:00:00 2001
From: FlyLoongZ <60963681+FlyLoongZ@users.noreply.github.com>
Date: Sat, 12 Apr 2025 14:43:59 +0800
Subject: [PATCH] Modify kernel version logic

---
 scripts/setlocalversion | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 842936656b846..b0edce2722add 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -91,15 +91,15 @@ scm_version()
 			# --abbrev=12 ensures a minimum of 12, and the
 			# awk substr() then picks the 'g' and first 12
 			# hex chars.
-			if atag="$(git describe --abbrev=12 2>/dev/null)"; then
-				echo "$atag" | awk -F- '{printf("-%05d-%s", $(NF-1),substr($(NF),0,13))}'
+			#####if atag="$(git describe --abbrev=12 2>/dev/null)"; then
+			##	echo "$atag" | awk -F- '{printf("-%05d-%s", $(NF-1),substr($(NF),0,13))}'
 
 			# If we don't have a tag at all we print -g{commitish},
 			# again using exactly 12 hex chars.
-			else
+			##else
 				head="$(echo $head | cut -c1-12)"
-				printf '%s%s' -g $head
-			fi
+				printf '%s%05d%s%s' - $(( ( $(od -An -N1 -tu1 < /dev/urandom) % 113 ) + 16 )) -g $head
+			#####fi
 		fi
 
 		# Is this git on svn?
@@ -113,12 +113,12 @@ scm_version()
 		# it fails. Note that git-diff-index does not refresh the
 		# index, so it may give misleading results. See
 		# git-update-index(1), git-diff-index(1), and git-status(1).
-		if {
-			git --no-optional-locks status -uno --porcelain 2>/dev/null ||
-			git diff-index --name-only HEAD
-		} | grep -qvE '^(.. )?scripts/package'; then
-			printf '%s' -dirty
-		fi
+		#####if {
+		##	git --no-optional-locks status -uno --porcelain 2>/dev/null ||
+		##	git diff-index --name-only HEAD
+		##} | grep -qvE '^(.. )?scripts/package'; then
+		##	printf '%s' -dirty
+		#####fi
 
 		# All done with git
 		return
-- 
2.49.0